gradio/chat_interface.py

```diff
@@ -0,0 +1,269 @@
+            title: a title for the interface; if provided, appears above chatbot in large font. Also used as the tab title when opened in a browser window.
+            description: a description for the interface; if provided, appears above the chatbot and beneath the title in regular font. Accepts Markdown and HTML content.
+            theme: Theme to use, loaded from gradio.themes.
+            css: custom css or path to custom css file to use with interface.
+            analytics_enabled: Whether to allow basic telemetry. If None, will use GRADIO_ANALYTICS_ENABLED environment variable if defined, or default to True.
+            submit_btn: Text to display on the submit button. If None, no button will be displayed. If a Button object, that button will be used.
+            retry_btn: Text to display on the retry button. If None, no button will be displayed. If a Button object, that button will be used.
+            delete_last_btn: Text to display on the delete last button. If None, no button will be displayed. If a Button object, that button will be used.
+            clear_btn: Text to display on the clear button. If None, no button will be displayed. If a Button object, that button will be used.
+        """
+        super().__init__(
+            analytics_enabled=analytics_enabled,
+            mode="chat_interface",
+            css=css,
+            title=title or "Gradio",
+            theme=theme,
+        )
+        self.fn = fn
+        self.examples = examples
+        self.cache_examples = cache_examples
+        self.buttons: list[Button] = []
+
+        with self:
+            if title:
+                Markdown(
+                    f"<h1 style='text-align: center; margin-bottom: 1rem'>{self.title}</h1>"
+                )
+            if description:
+                Markdown(description)
+
+            with Group():
+                if chatbot:
+                    self.chatbot = chatbot.render()
+                else:
+                    self.chatbot = Chatbot(label="Input")
+                if textbox:
+                    self.textbox = textbox.render() 
+                else: 
+                    self.textbox = Textbox(show_label=False, placeholder="Type a message...")
+            
+            with Row():
+                if submit_btn:
+                    if isinstance(submit_btn, Button):
+                        submit_btn.render()
+                    elif isinstance(submit_btn, str):
+                        submit_btn = Button(submit_btn, variant="primary")
+                    else:
+                        raise ValueError(f"The submit_btn parameter must be a gr.Button, string, or None, not {type(submit_btn)}")
+                    self.buttons.append(submit_btn)
+                self.stop_btn = Button("Stop", variant="stop", visible=False)
+                
+                for btn in [retry_btn, delete_last_btn, clear_btn]:
```

Comment:
nit: why not use a gr.ClearButton here?

Reply:
Sure, will update

Reply:
Hmm it actually complicates things a bit because we have to separately handle the case if users supply a button (in which case we still have to add the event handlers manually). I'll just leave as is